### PR TITLE
Check for permission upgrade

### LINKED ISSUE #1614: more collaborators tests
scenarios copied from old webUI does not work, because the workflow changed see https://github.com/owncloud/enterprise/blob/master/product/oc10/index.md#collaborators
Some tests were implemented, but by far not enough.
missing e.g.
- resharing
- changing collaborators type - #2124
- allow / disallow resharing
- unshare as receiver/sender
- etc.

### PR DESCRIPTION
**Pull request by skshetry**
## Description

Testing permission upgrades in Phoenix.

## Related Issue

- Fixes #1614

## How Has This Been Tested?

- locally and :robot:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Trashbin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/5734/

<p>



















</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPermissions</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/5734/

<p>





















</p></details>

**Comment by HariBhandari07** [comment]
LGTM

File: tests/acceptance/features/webUISharingPermissionsUsers/sharePermissionsUsers.feature [hunk 1]
```diff
@@ -264,3 +264,52 @@ Feature: Sharing files and folders with internal users with different permission
     When the user opens the share dialog for folder "simple-folder" using the webUI
     Then user "User One" should be listed as "Viewer" in the collaborators list for folder "simple-folder" on the webUI
     And no custom permissions should be set for collaborator "User One" for folder "simple-folder" on the webUI
+
+  Scenario: User permissions are upgraded
```
Comment (individual-it) [inline tests/acceptance/features/webUISharingPermissionsUsers/sharePermissionsUsers.feature]:
```suggestion
  Scenario: Resource owner upgrades share permissions of a re-share
```

File: tests/acceptance/features/webUISharingPermissionsUsers/sharePermissionsUsers.feature [hunk 2]
```diff
@@ -264,3 +264,52 @@ Feature: Sharing files and folders with internal users with different permission
...
+  Scenario: User is not allowed to reshare sub-folder with more permissions
+    Given user "user3" has been created with default attributes
+    And user "user2" has shared folder "simple-folder" with user "user1" with "read, share, delete" permissions
+    And user "user1" has logged in using the webUI
+    When the user browses to the folder "simple-folder (2)" on the files page
+    And the user shares folder "simple-empty-folder" with user "User Three" as "Custom role" with permissions "share, delete, change" using the webUI
+    Then the error message "Error while sharing." should be displayed on the webUI
+    And as "user3" folder "simple-empty-folder (2)" should not exist
+
+  Scenario: User is not allowed to update reshare of sub-folder with more than what user has
```
Comment (individual-it) [inline tests/acceptance/features/webUISharingPermissionsUsers/sharePermissionsUsers.feature]:
```suggestion
  Scenario: User is not allowed to update permissions of a reshared sub-folder to higher permissions than the own ones
```

File: tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js [hunk 3]
```diff
@@ -40,6 +40,27 @@ module.exports = {
       )
     },
 
+    /**
+     * Return first elementID that matches given selector and is visible
+     *
+     * @param {string} using
+     * @param {string} value
+     */
+    getVisibleElementID: async function (using, value) {
```
Comment (individual-it) [inline tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js]: would that be useful also in other places? maybe put it into a custom command?

Comment (skshetry) [reply tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js]: `CustomCommands` need to return itself. So, we would need to use the callback function instead. If we need it elsewhere, we can think of it then(following YAGNI).

**Comment by HariBhandari07** [comment]
LGTM

File: tests/acceptance/features/webUISharingPermissionsUsers/sharePermissionsUsers.feature [hunk 4]
```diff
@@ -146,3 +146,52 @@ Feature: Sharing files and folders with internal users with different permission
...
+    Then the error message "Error while sharing." should be displayed on the webUI
+    And as "user3" folder "simple-empty-folder (2)" should not exist
+
+  Scenario: User is allowed to update reshare of sub-folder with more than what user has
+    Given user "user3" has been created with default attributes
+    And user "user2" has shared folder "simple-folder" with user "user1" with "all" permissions
+    And user "user1" has shared folder "simple-folder (2)" with user "user3" with "share, delete" permissions
+    And user "user1" has logged in using the webUI
+    When the user browses to the folder "simple-folder (2)" on the files page
+    Then the user shares folder "simple-empty-folder" with user "User Three" as "Custom role" with permissions "share, delete, create, change" using the webUI
```
Comment (phil-davis) [inline tests/acceptance/features/webUISharingPermissionsUsers/sharePermissionsUsers.feature]:
```suggestion
    And the user shares folder "simple-empty-folder" with user "User Three" as "Custom role" with permissions "share, delete, create, change" using the webUI
```

File: tests/acceptance/features/webUISharingPermissionsUsers/sharePermissionsUsers.feature [hunk 5]
```diff
@@ -146,3 +146,52 @@ Feature: Sharing files and folders with internal users with different permission
...
+    And user "user3" should have received a share with these details:
```
Comment (phil-davis) [inline tests/acceptance/features/webUISharingPermissionsUsers/sharePermissionsUsers.feature]:
```suggestion
    Then user "user3" should have received a share with these details:
```

File: tests/acceptance/features/webUISharingPermissionsUsers/sharePermissionsUsers.feature [hunk 6]
```diff
@@ -146,3 +146,52 @@ Feature: Sharing files and folders with internal users with different permission
...
+    Given user "user3" has been created with default attributes
+    And user "user2" has shared folder "simple-folder" with user "user1" with "read, share, delete, change" permissions
+    And user "user1" has shared folder "simple-folder (2)" with user "user3" with "share, delete" permissions
+    And user "user1" has logged in using the webUI
+    When the user browses to the folder "simple-folder (2)" on the files page
+    And the user shares folder "simple-empty-folder" with user "User Three" as "Custom role" with permissions "share, delete, change, create" using the webUI
+    Then the error message "Error while sharing." should be displayed on the webUI
+    And as "user3" folder "simple-empty-folder (2)" should not exist
+
+  Scenario: User is allowed to update reshare of sub-folder with more than what user has
```
Comment (phil-davis) [inline tests/acceptance/features/webUISharingPermissionsUsers/sharePermissionsUsers.feature]:
```suggestion
  Scenario: User is allowed to update permissions of a reshared sub-folder within the permissions that the user has received
```

**Pull request merged**